[DOCS] add docs for async search (#53675)
elastic/elasticsearch: docs/reference/async-search.asciidoc

[role="xpack"]
[testenv="basic"]
[[async-search-intro]]

> == Long running searches

Reviewer: ```suggestion
== Long-running searches
```

> {es} generally allows to quickly search across big amounts of data. There are

Reviewer: ```suggestion
{es} generally allows you to quickly search across big amounts of data. There are
```

situations where a search executes on many many shards, possibly against

> <<frozen-indices>> and spanning multiple <<modules-remote-clusters>>, for which

Reviewer: We need to provide an anchor text. Otherwise, these will be capitalized.
```suggestion
<<frozen-indices,frozen indices>> and spanning multiple
<<modules-remote-clusters,remote clusters>>, for which
```

> results are not expected to be returned in milliseconds. When needing to

Reviewer: ```suggestion
results are not expected to be returned in milliseconds. When you need to
```

> execute long running searches, sending a search request and synchronously

Reviewer: ```suggestion
execute long-running searches, synchronously
```

waiting for its results to be returned is not ideal. Async Search allows
instead to submit a search request that gets executed asynchronously and

> monitor its progress as well as retrieve its results at a later stage.

Reviewer: ```suggestion
waiting for its results to be returned is not ideal. Instead, Async search lets
you submit a search request that gets executed _asynchronously_,
monitor the progress of the request, and retrieve results at a later stage.
```

Also, partial results can be retrieved as they become available, before

> the search has completed.

Reviewer: ```suggestion
You can also retrieve partial results as they become available but
before the search has completed.
```

An async search request can be submitted using the <<submit-async-search>> API.
The <<get-async-search>> API allows to monitor the progress of an async search
request and retrieve its results. An async search can be deleted through the

> <<delete-async-search>> API.

Reviewer: ```suggestion
You can submit tn async search request using the <<submit-async-search,submit async search>> API.
The <<get-async-search,get async search>> API allows you to monitor the progress of an async search
request and retrieve its results. An ongoing async search can be deleted through the
<<delete-async-search,delete async search>> API.
```
